$NetBSD: patch-bt,v 1.2 2011/07/19 07:54:31 manu Exp $

Fix the io-threads xlator, which forgot to initialize its pthread
condition and mutex.

--- xlators/performance/io-threads/src/io-threads.c.orig
+++ xlators/performance/io-threads/src/io-threads.c
@@ -2181,8 +2181,20 @@
                         "out of memory");
                 goto out;
         }
 
+	if ((ret = pthread_cond_init(&conf->cond, NULL)) != 0) {
+                gf_log (this->name, GF_LOG_ERROR,
+                        "pthread_cond_init failed (%d)", ret);
+                goto out;
+	}
+
+	if ((ret = pthread_mutex_init(&conf->mutex, NULL)) != 0) {
+                gf_log (this->name, GF_LOG_ERROR,
+                        "pthread_mutex_init failed (%d)", ret);
+                goto out;
+	}
+
         set_stack_size (conf);
 
         thread_count = IOT_DEFAULT_THREADS;
 
